src/utils/parse_heading.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+/**
+ * Interface describing a parsed heading result.
+ *
+ * @private
+ */
+interface IHeading {
+  /**
+   * Heading text.
+   */
+  text: string;
+
+  /**
+   * Heading level.
+   */
+  level: number;
+
+  /**
+   * Heading type.
+   */
+  type: 'html' | 'markdown' | 'markdown-alt';
```

Comment:
Interesting. What is the purpose of differentiating based on these different styles of markdown heading?

Reply:
Because this logic is moved to its own utility function, needs to be generic enough to handle, e.g., the case where you parse a string for a heading, but then, based on the heading type, want to do further manipulation. That manipulation may depend on what type of heading was recovered. For alternative Markdown formatting, that might include removing the second line and replacing the first with `#####` style headings, etc. So, in short, as I plucked this logic from other contexts, I opted for retaining the "type" of the heading recovered in the spirit of a general utility.